Tasks/Npm/npmtoolrunner.ts

```diff
@@ -74,4 +86,47 @@ export class NpmToolRunner extends tr.ToolRunner {
         let config = tl.execSync('npm', `config list ${this.dbg ? '-l' : ''}`, options);
         return options;
     }
+
+    private _getDebugLogPath(options?: tr.IExecSyncOptions): string {
+        // check cache
+        options.silent = true;
+        const cacheLocation = tl.execSync('npm', 'config get cache', options).stdout.trim();
```

Comment:
Yes, some versions will clear out either the npm-debug.log in the working directory, or whatever the log file path would have been in the cache. I'll move this up so we get the settings on the ToolRunner creation.